### DISCUSSION TITLE
Best Ollama language model?

### DISCUSSION THREAD

**Discussion by tehniemer** [karakeep-app/karakeep#430] [Q&A]
Does anyone have a recommendation on which Ollama language model works best for Hoarder?

**Comment by kamtschatka** [comment]
i don't know if it works best, but I get good results with this:
```
INFERENCE_TEXT_MODEL=llama3
INFERENCE_IMAGE_MODEL=llava
INFERENCE_LANG=english
```

**Reply by tehniemer** [reply]
I imported about 1800 bookmarks and the inference jobs failed on all of them, I'm not sure why. I changed the models, but now I don't see a way to run the inference again.

**Reply by volteret4** [reply]
I just installed Hoarder, but in the ADMIN tab you have some ways to run the inference again, on failed links only or all of your links.

Is that what you are looking for? Probably I didn't understand you either.

**Comment by Snuupy** [comment]
I'm using `x/llama3.2-vision:11b-instruct-q8_0` but I get a lot of new tags and not a lot of repeated/categorized tags

for example, I see: self-hosting and self-hostable software as separate tags, but they really should just be under a single tag: self-hosted

maybe a lower parameter 1B-3B model would be better so it has less vocabulary to use 😂, or that vector db https://github.com/hoarder-app/hoarder/issues/267

**Comment by xxjoe2** [comment]
i am a newbie. i just installed llama3.2 3b but i dont think its good enough.

i provided it with a web page and asked it to generate tags for me but it showed me those titles, sub titles and a brief summary etc but not tags

also tried gpt 4o mini fro  llama but the results were not what expected. sometimes it game me pretty good tags, but sometimes it replied something like "the height of buildings". whats more is it reponses hard coded ads like things about the author in a few occasions

**Comment by fsiler** [comment]
on CPU only (NUC12 i5), I'm having good luck using `qwen2.5:3b`.  I gave the following additional instructions to try to normalize tags a bit:

- Use capital letters for proper nouns; otherwise lowercase.
- Use spaces to separate words.
- Do not include cookies or privacy policy unless the article is about implementation of these
- Do not use underscores or dashes unless they are part of a proper name.

**Comment by rymurr** [comment]
I've been using [gemma3 1b](https://ollama.com/library/gemma3/tags) on CPU and its been very good.
